### DISCUSSION TITLE
TabularPredictor: Are bagged models saved in the order of `groups`?

### DISCUSSION THREAD

**Discussion by dluks** [autogluon/autogluon#4368] [Q&A] [answered]
I am manually specifying `groups` when fitting a `TabularPredictor`. For example, I have 10 groups (indicated by values 1-10 in the `groups` column of my training data), and after fitting, I can see 10 models named `S1F1`, `S1F2`, `S1F3`, ..., etc.

Does Autogluon process each group in order? In other words, can I reliably know that the `S1F1` model was trained on groups 2-10, with the holdout fold being group 1?

I've looked through the code to figure this out but it's a bit hard to follow.

**Comment by Innixma** [comment, marked as answer]
The split logic in AutoGluon works like this with `groups`:

```
from sklearn.model_selection import LeaveOneGroupOut
split_order_indices = [[train_index, test_index] for train_index, test_index in LeaveOneGroupOut().split(X, y, groups=groups)]
```

So it functions as sklearn implemented [LeaveOneGroupOut](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.LeaveOneGroupOut.html).

The docs say:

> Splits are ordered according to the index of the group left out. The first split has testing set consisting of the group whose index in groups is lowest, and so on.

Therefore, the logic aligns with your description.
